app/structures/weighted_adjacency_list.py

```diff
@@ -338,4 +338,13 @@ def convert_to_networkX(self) -> nx.DiGraph:
             for end_vertex in self.graph[start_index]:
                 visualization.add_edge(start_index, end_vertex.index, weight=end_vertex.weight)
         return visualization
-            
\ No newline at end of file
+
+    
+    def convert_to_matrix(self):
```

Comment:
I think it could be "to_adjacency_matrix()" method which returns AdjacencyMatrix object (it requires simple changes in AdjacencyMatrix methods)